editor/src/components/canvas/canvas-utils.ts

```diff
@@ -2816,14 +2818,33 @@ export function duplicate(
               }
             }
 
-            const insertResult = insertElementAtPath_DEPRECATED(
-              workingEditorState.projectContents,
-              workingEditorState.canvas.openFile?.filename ?? null,
-              optionalMap(childInsertionPath, newParentPath),
+            const conditionalCase = getConditionalCaseCorrespondingToBranchPath(
+              path,
+              editor.jsxMetadata,
+            )
+
+            if (conditionalCase != null && isEmptyConditionalBranch(path, editor.jsxMetadata)) {
+              // can't duplicate empty conditional branch
+              return success
+            }
+
+            const insertionPath =
+              conditionalCase != null
+                ? conditionalClauseInsertionPath(
+                    EP.parentPath(path),
+                    conditionalCase,
+                    'wrap-with-fragment',
+                  )
+                : childInsertionPath(EP.parentPath(newPath))
```

Comment:
I think you could refactor this to benefit from `getInsertionPathWithWrapWithFragmentBehavior`